ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -38,33 +38,14 @@ let update_mh_info interface =
 
 let stunnel_m = Mutex.create ()
 
-let restart_stunnel_nomutex ~__context ~accept =
+let restart_stunnel ~accept =
   info "Restarting stunnel (accepting connections on %s)" accept;
-  let xapissl_args = [ "restart"
-                     ; accept
-                     ; "permfile=" ^ !Xapi_globs.server_cert_path
-                     ; Xcp_const.good_ciphersuites
-                     ]
-  in
-  let (_ : Thread.t) = Thread.create (fun () ->
-      Mutex.execute management_m (fun () ->
-          Forkhelpers.execute_command_get_output !Xapi_globs.xapissl_path xapissl_args
-        )
-    ) () in
-  ()
+  Helpers.Stunnel.restart ~accept
```

Comment:
This previously used to be a thread, and now is inline, it might be better to keep it in a thread